concordium-sdk/src/main/java/com/concordium/sdk/cis2/SerializationUtils.java

```diff
@@ -0,0 +1,297 @@
+import lombok.SneakyThrows;
+import lombok.val;
+
+import java.io.ByteArrayOutputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class SerializationUtils {
+
+    /**
+     * Size of a serialized {@link AccountAddress}
+     * Tag + the bytes of the account address.
+     */
+    private static final int ACCOUNT_ADDRESS_SIZE = 1 + AccountAddress.BYTES;
+
+    /**
+     * Size of a serialized {@link ContractAddress}
+     * Tag + 8 bytes for the {@link ContractAddress#getIndex()} + 8 bytes for the {@link ContractAddress#getSubIndex()}.
+     */
+    private static final int CONTRACT_ADDRESS_SIZE = 1 + (2 * 8);
+
+    @SneakyThrows
+    public static Parameter serializeTransfers(List<Cis2Transfer> listOfTransfers) {
+        val bos = new ByteArrayOutputStream();
+        bos.write(UInt16.from(listOfTransfers.size()).getBytesLittleEndian());
+        for (Cis2Transfer transfer : listOfTransfers) {
+            bos.write(SerializationUtils.serializeTokenId(transfer.getTokenId()));
+            writeUnsignedLeb128(bos, transfer.getTokenAmount());
+            bos.write(SerializationUtils.serializeAddress(transfer.getSender()));
+            bos.write(SerializationUtils.serializeAddress(transfer.getReceiver()));
+            if (Objects.isNull(transfer.getAdditionalData()) || transfer.getAdditionalData().length == 0) {
+                bos.write(UInt16.from(0).getBytesLittleEndian());
+            } else {
+                bos.write(UInt16.from(transfer.getAdditionalData().length).getBytesLittleEndian());
+                bos.write(transfer.getAdditionalData());
+            }
+        }
+        return Parameter.from(bos.toByteArray());
+    }
+
+    @SneakyThrows
+    static Parameter serializeBalanceOfParameter(Collection<BalanceQuery> queries) {
+        val bos = new ByteArrayOutputStream();
+        // lengths are stored as little endian.
+        bos.write(UInt16.from(queries.size()).getBytesLittleEndian());
+        for (BalanceQuery balanceQuery : queries) {
+            bos.write(SerializationUtils.serializeTokenId(balanceQuery.getTokenId()));
+            bos.write(SerializationUtils.serializeAddress(balanceQuery.getAddress()));
+        }
+        return Parameter.from(bos.toByteArray());
+    }
+
+    static long[] deserializeTokenAmounts(byte[] returnValue) {
+        val resultBuffer = ByteBuffer.wrap(returnValue);
+        // lengths are stored as little endian.
+        resultBuffer.order(ByteOrder.LITTLE_ENDIAN);
+        val noOfOutputs = UInt16.from(resultBuffer.getShort()).getValue();
+        resultBuffer.order(ByteOrder.BIG_ENDIAN);
+        val outputs = new long[noOfOutputs];
+        for (int i = 0; i < noOfOutputs; i++) {
+            outputs[i] = readUnsignedLeb128(resultBuffer);
```

Comment:
We also limit the amount of bytes to I think 37 or some such. It should be in the standard. Not sure if you want to enforce this here or not.

Reply:
Added a check.